src/ui/i18n/i18n.js

```diff
@@ -0,0 +1,107 @@
+import path from 'path';
+import Promise from 'bluebird';
+import { readFile } from 'fs';
+import _ from 'lodash';
+
+const asyncReadFile = Promise.promisify(readFile);
+
+const TRANSLATION_FILE_EXTENSION = '.json';
+
+let registeredTranslations = {};
```

Comment:
Good point about the `registeredTranslations`.  :thumbsup:

I am wondering though if it would be better to maintain all localization within `i18n` and keep the `UiI18n` class for handling the interaction between the UI/server and localization?

With that in mind, I have updated the `i18n` module into a class called `I18n`. This makes it possible to have localization at server level as `UiI18n` instance now contains an instance of the `I18n` class. I think it also better to encapsulate localization within the `I18n` class. In the process I have tidied up `_build:verifyTranslations` and removed its dependency on `i18n`. I have also refactored the unit tests and removed the test build up and tear down. Let me know what you think.

There are 3 methods in the `I18n`class (`getRegisteredTranslationLocales`, `getTranslationLocale` and `getTranslationsForLocale`) which I would like to make private but not sure how this is done in JS. Any suggestions are welcome!  :smile: